mod/Jailbreak.Rebel/JihadC4/JihadC4Behavior.cs

```diff
@@ -0,0 +1,291 @@
+﻿using CounterStrikeSharp.API;
+using CounterStrikeSharp.API.Core;
+using CounterStrikeSharp.API.Core.Attributes.Registration;
+using CounterStrikeSharp.API.Modules.Commands;
+using CounterStrikeSharp.API.Modules.Memory.DynamicFunctions;
+using CounterStrikeSharp.API.Modules.Utils;
+using Jailbreak.Formatting.Extensions;
+using Jailbreak.Formatting.Views;
+using Jailbreak.Public.Behaviors;
+using Jailbreak.Public.Extensions;
+using Jailbreak.Public.Mod.Rebel;
+
+namespace Jailbreak.Rebel.JihadC4;
+
+public class JihadC4Behavior : IPluginBehavior, IJihadC4Service
+{
+
+    // Importantly the Player argument CAN be null!
+    private class JihadBombMetadata(CCSPlayerController? player, float delay) { public CCSPlayerController? Player { get; set; } = player; public float Delay { get; set; } = delay; }
+    // Key presents any active Jihad C4 in the world. Values represent metadata about that Jihad C4.
+    private Dictionary<CC4, JihadBombMetadata> _currentActiveJihadC4s = new();
+
+    private IJihadC4Notifications _jihadNotifications;
+    private BasePlugin? _basePlugin;
+
+    // EmitSound(CBaseEntity* pEnt, const char* sSoundName, int nPitch, float flVolume, float flDelay)
+    private readonly MemoryFunctionVoid<CBaseEntity, string, int, float, float> CBaseEntity_EmitSoundParamsLinux; // LINUX ONLY.
+
+    public JihadC4Behavior(IJihadC4Notifications jihadC4Notifications)
+    {
+        _jihadNotifications = jihadC4Notifications;
+        CBaseEntity_EmitSoundParamsLinux = new("48 B8 ? ? ? ? ? ? ? ? 55 48 89 E5 41 55 41 54 49 89 FC 53 48 89 F3");
```

Comment:
prefer to see this implemented for the entire gamemode to use (ex warden sounds) but that can wait